### DISCUSSION TITLE
Allow testing on localhost

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#445] [Anything Else]
"In some cases, single person/small teams might set things up on localhost first and from there do all their building/testing, instead of on a dedicated staging site. And so with a free trial, it would be nice to be able to see how the product would work on localhost before deciding whether or not to go with it. Also, testing out custom events from localhost would be nice too. Something like: `plausible('config', { debug: true });`"

(feedback via email)

**UPDATE:** We now have the plausible.local.js extension that allows you to track stats on localhost too. [Read more here](https://plausible.io/docs/script-extensions). Thanks @riderx for this contribution!

**Comment by JH7** [comment]
First things first: Thanks for creating Plausible, it's been a pleasure to use for our web app project so far!

We are currently running into this "localhost problem" from a different perspective. We use [Meteor](https://github.com/meteor/meteor/) and [Cordova](https://github.com/apache/cordova) to bake our existing Meteor web app into a native app. In the native app, web pages are stored on the device and then served from localhost. Thus, we are currently not able to track interactions in the native app.

A possible solution to that could be the adding of another tracker script mode.
https://github.com/plausible/analytics/blob/ff32218bd030f28b0d9ee80dc39952a190d8faab/tracker/src/plausible.js#L20
This line could then be enclosed e.g. by a ``{{#if localhostMode}}``.

**Reply by ukutaht** [reply]
Agreed. Sounds like there's no other way for you to use Plausible. Making a localhost mode is probably the best solution.

**Reply by Sha-code** [reply]
We have the same issue for our apps https://Captime.app (angular capacitor) and https://mimesis.fun (react capacitor), just a localhost mode will make it works for us !

**Reply by riderx** [reply]
Same here i need it for capacitor app (Cordova like)

**Reply by riderx** [reply]
@ukutaht @metmarkosaric do you think it can came to the product soon ?

**Reply by riderx** [reply]
I did a Pr for that ! https://github.com/plausible/analytics/pull/1218
I hope it will helps !

**Comment by metmarkosaric** [comment]
We now have the plausible.local.js extension that allows you to track stats on localhost too. [Read more here](https://plausible.io/docs/script-extensions). Thanks @riderx for this contribution!

**Reply by alvarlagerlof** [reply]
You should probably mark this as an answer

**Comment by elron** [comment]
Just for better understanding,
should I have two separated plausable sites, one for development (script.local.js) and one for production (script.js)?
